test/integration_test.tsx

```diff
@@ -761,35 +788,230 @@ describe('integration tests', () => {
       assert.equal(wrapper.text(), 'Example');
     });
 
-    it('supports simulating events on Components (simulateEventsOnComponents: true)', () => {
-      function FancyButton({ onClick, children }: any) {
-        return (
-          <button type="button" onClick={onClick}>
-            {children}
-          </button>
+    describe('simulateEventsOnComponents: true', () => {
+      setAdapter(() => new Adapter({ simulateEventsOnComponents: true }));
+
+      it('supports simulating events on Components', () => {
+        function FancyButton({ onClick, children }: any) {
+          return (
+            <button type="button" onClick={onClick}>
+              {children}
+            </button>
+          );
+        }
+
+        function App() {
+          const [count, setCount] = useState(0);
+
+          return (
+            <div>
+              <div id="count">Count: {count}</div>
+              <FancyButton onClick={() => setCount(count + 1)}>
+                Increment
+              </FancyButton>
+            </div>
+          );
+        }
+
+        const wrapper = shallow(<App />);
+        assert.equal(wrapper.find('#count').text(), 'Count: 0');
+
+        wrapper.find(FancyButton).simulate('click');
+        assert.equal(wrapper.find('#count').text(), 'Count: 1');
+      });
+    });
+
+    describe('preserveFragmentsInShallowRender: true', () => {
+      setAdapter(() => new Adapter({ preserveFragmentsInShallowRender: true }));
+
+      addStaticTests(shallow);
+      addInteractiveTests(shallow as any);
+
+      it('supports update and setState on Components that return Fragments with multiple children', () => {
```

Comment:
FYI, this is the test that motivated this change. I wanted to have `update` and `setState` work on shallow rendered components that returned Fragments.